openapi/nexus.json

```diff
@@ -13706,22 +13706,37 @@
         ]
       },
       "Sled": {
-        "description": "Client view of a [`Sled`]",
+        "description": "An operator's view of a sled.",
         "type": "object",
         "properties": {
           "baseboard": {
             "$ref": "#/components/schemas/Baseboard"
           },
+          "cpus": {
+            "description": "The number of online processors available to the sled.",
+            "type": "integer",
+            "format": "int64"
+          },
           "id": {
             "description": "unique, immutable, system-controlled identifier for each resource",
             "type": "string",
             "format": "uuid"
           },
+          "physical_ram": {
+            "description": "The total amount of physical RAM available to the sled.",
+            "allOf": [
+              {
+                "$ref": "#/components/schemas/ByteCount"
+              }
+            ]
+          },
           "rack_id": {
+            "description": "The rack to which this Sled is currently attached.",
```

Comment:
I think we may want to think carefully about the term rack_id as we have no good way of actually relating this to a physical rack. It's really a logical control plane instance id for a rack.

I mention this because this suggests that it can change due to moving the sled, but if we move all the sleds from rack A to rack B, it won't change. So I think we may want to be more precise. This may be something that should come up as a different follow up along with the bit below.

Reply:
Meghan actually if there was something we needed to do during manufacturing or setup to include information from the rack's nameplate.

@smklein this came up during the RFD-267 huddle too. Essentially it seems like we're going to be mostly unable to tell what sled occupies which rack (accurately, anyway). I believe this is going to be fairly problematic in a multi-rack deployment in that it both breaks some of the assumptions in the relational hierarchy and it'll also be difficult to communicate to an operator the physical location of a piece of hardware in their overall installation.